src/doc.d

```diff
@@ -2479,6 +2499,73 @@ extern (C++) void highlightCode(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 i = j - 1;
             }
         }
+        else if (!resolvedTemplateParameters)
+        {
+            // hunt for template declarations:
+            for (size_t symi = 0; symi < a.dim; symi++)
+            {
+                FuncDeclaration fd = (*a)[symi].isFuncDeclaration();
+
+                if (!fd || !fd.parent || !fd.parent.isTemplateDeclaration())
+                {
+                    continue;
+                }
+
+                TemplateDeclaration td = fd.parent.isTemplateDeclaration();
+
+                // build the template parameters
+                size_t[] paramLens;
+                OutBuffer parametersBuf;
+                HdrGenState hgs;
+                parametersBuf.writeByte('(');
+
+                for (size_t parami = 0; parami < td.parameters.dim; parami++)
+                {
+                    TemplateParameter tp = (*td.parameters)[parami];
+
+                    if (parami)
+                        parametersBuf.writestring(", ");
+
+                    size_t lastOffset = parametersBuf.offset;
+
+                    .toCBuffer(tp, &parametersBuf, &hgs);
+
+                    paramLens ~= [parametersBuf.offset - lastOffset];
+                }
+                parametersBuf.writeByte(')');
+
+                const(char*) templateParams = parametersBuf.extractString();
+                size_t templateParamsLen = strlen(templateParams);
+
+                //printf("templateDecl: %s\ntemplateParams: %s\nstart: %s\n", td.toChars(), templateParams, start);
+
+                if (cmp(templateParams, start, templateParamsLen) == 0)
+                {
+                    const(char*) templateParamListMacro = "$(DDOC_TEMPLATE_PARAM_LIST ";
```

Comment:
Being a string literal, the data is already static (and immutable most the time). If you make the pointer static too, it, at best, makes no difference and likely has a (really small) negative effect on the generated binary.

Static sometimes makes sense with D array literals because then you can avoid a runtime allocation with them. But string literals are never runtime allocated anyway. Adding static here just pads the code unless I'm missing something.